### PR TITLE
DROOLS-3625: [DMN Designer] Remove Stunners Explorer widget from the RHS dock

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.jboss.org/browse/DROOLS-3625

This PR adds the ability for different consumers of _Stunner_ to provide their own "Authoring Docks" implementation. BPMN and CM use the existing one (Properties Panel dock, Explorer dock with preview and explorer widgets). DMN now has a custom one containing Properties Panel dock and a new Explorer dock containing just the preview widget).

I've also removed some classes that were not used anywhere.

### CODE REVIEW

File: kie-wb-common-stunner/kie-wb-common-stunner-extensions/kie-wb-common-stunner-project/kie-wb-common-stunner-project-client/src/main/java/org/kie/workbench/common/stunner/project/client/docks/StunnerDocksHandler.java [hunk 1]
```diff
@@ -16,56 +16,66 @@
...
-    public void onDiagramEditorMaximized(@Observes ScreenMaximizedEvent event) {
+    public void onDiagramEditorMaximized(final @Observes ScreenMaximizedEvent event) {
         if (event.isDiagramScreen()) {
             refreshDocks(true,
                          false);
         }
     }
+
+    private StunnerDockSupplier doLookup(final Annotation[] qualifiers) {
+        final ManagedInstance<StunnerDockSupplier> i = dockSuppliers.select(qualifiers);
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-extensions/kie-wb-common-stunner-project/kie-wb-common-stunner-project-client/src/main/java/org/kie/workbench/common/stunner/project/client/docks/StunnerDocksHandler.java]: probably you can just reuse some method from [InstanceUtils](https://github.com/kiegroup/kie-wb-common/blob/master/kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/session/impl/InstanceUtils.java#L26) here

Comment (manstis) [reply kie-wb-common-stunner/kie-wb-common-stunner-extensions/kie-wb-common-stunner-project/kie-wb-common-stunner-project-client/src/main/java/org/kie/workbench/common/stunner/project/client/docks/StunnerDocksHandler.java]: FYI @romartin @jomarko @hasys I've made Roger's change in a new commit.

Comment (jomarko) [reply kie-wb-common-stunner/kie-wb-common-stunner-extensions/kie-wb-common-stunner-project/kie-wb-common-stunner-project-client/src/main/java/org/kie/workbench/common/stunner/project/client/docks/StunnerDocksHandler.java]: Rechecked. Still green from me.

**Comment by manstis** [comment]
@hasys Could you please review when you get a chance. Thank-you.

**Comment by manstis** [comment]
Jenkins please retest this.

File: kie-wb-common-stunner/kie-wb-common-stunner-extensions/kie-wb-common-stunner-project/kie-wb-common-stunner-project-client/src/main/java/org/kie/workbench/common/stunner/project/client/docks/StunnerDocksHandler.java (hunk 1 above)
Comment (romartin) [reply kie-wb-common-stunner/kie-wb-common-stunner-extensions/kie-wb-common-stunner-project/kie-wb-common-stunner-project-client/src/main/java/org/kie/workbench/common/stunner/project/client/docks/StunnerDocksHandler.java]: @jomarko it's fine, it says `outdated` on top.. I checked the second commit and new line is there :+1:

**Comment by manstis** [comment]
@romartin @jomarko Do we need @hasys to review this (I thought so, for QE coverage of BPMN).

However he neither appears to be on PTO or respond to my requests for review.

Can I remove him and merge this PR?!

**Comment by romartin** [comment]
Hey @manstis 

Yeah noticed @hasys @LuboTerifaj  are not around.... dunno :-( 

AFAIK in that "edge" cases, where any engineering task (PRs) is waiting QE review, but it's not being done, (for whatever reason), we are able to delegate the PR review to any other guy from engineering and go ahead with the merge when approved... Anyway better letting QE know about this action (for further testing) and also your project manager (Edson?). 

So as this PR is already approved by me and @jomarko ,  I think it's fine to go ahead yeah... If all goes well, I'll feel proud otherwise will say it's your fault :-P

**Comment by jomarko** [comment]
According my acceptance test, see https://issues.jboss.org/browse/DROOLS-3625, I checked that bpmn still has the touched component, I think it is fine to merge. @hasys has PTO and will be back on the 18th of June.

**Pull request merged**